Add sleepy advertiser example

ahedin25 opened the pull request:
This application can be used to measure sleep current when advertising
or in power off mode.

Modified from zephyr/samples/boards/nrf/system_off and
zephyr/samples/bluetooth/beacon.

Signed-off-by: Andrew Hedin <andrew.hedin@lairdconnect.com>

rerickson1 commented on sleepy_advertiser/Kconfig:
```diff
@@ -0,0 +1,36 @@
+	bool "Advertise on startup"
+	depends on BT
+	help
+	  Rate is approximately 1 second.
```
Maybe be a little more verbose here. Does this mean the advertisement rate? Does it advertise forever?

ahedin25 committed:
sleepy_advertiser: Add example

This application can be used to measure sleep current when advertising
or in power off mode.

Modified from zephyr/samples/boards/nrf/system_off and
zephyr/samples/bluetooth/beacon.

Signed-off-by: Andrew Hedin <andrew.hedin@lairdconnect.com>
